www/js/splash/pushnotify.js

```diff
@@ -15,12 +15,13 @@
 
 import angular from 'angular';
 import { updateUser } from '../commHelper';
+import { readConsentState, isConsented, startPrefs } from './startprefs';
+import { readIntroDone } from '../onboarding/onboardingHelper';
```

Comment:
see, this is actually a circular dependency. onboardingHelper depends on pushnotify which depends on onboardingHelper. Is that not a problem/considered bad practice in react?

Reply:
I haven't done extensive research into the way react wants us to handle things like this, but the good news here is that is an old dependency, so line 19 can just be removed. 

First, we had the event handler - then I rewrote to call the same code (check for intro done then call `registerPush`

With Jack's help in review, this got cleaner https://github.com/e-mission/e-mission-phone/pull/1072#discussion_r1372339180. Now, the code that was called by the "intro done" event is called after intro is marked in `onboardingHelper` and the "reconsent" event code is called in from `startprefs` (I commented on these places above to flag them).

So there is no circular dependency, since we only call the "actionable" code from these other files once we've already checked the conditions, this was a major improvement during Jack's review.

I will do some more research about React's philosophy on tight dependencies and possibilities for event driven implementation.

Reply:
There is still a good bit of circular dependency, as there is two-way dependency between:

`startPrefs` and each of `pushNotify`, `onboardingHelper`, and `storeDeviceSettings`

In each case, the dependency on `startPrefs` is for `readConsentState` and `isConsented`. I'm wondering if the more React way to handle that dependency would be to use `appContext`? But, at least the way we've used it so far it only applies to components. 

It might make some sense to collapse `startPrefs` into `onboardingHelper`, since both work some with consent, but that won't fix this problem.

Reply:
Most of what I've been able to find about tight coupling or event emitting suggests passing the event handler into the child component, or using context as the "React" way of doing things. I have not found much pertaining specifically to modules like these. @JGreenlee do you know what the "React" way to navigate this kind of circular dependency should be of if we can use `appContext` to help with this?

Reply:
I started an issue [here](https://github.com/e-mission/e-mission-docs/issues/1024), as I've kept looking into this and it seems like it will be a much bigger deal when it comes to the "notify" family of services

Reply:
I think that we will be circling back to adding in events, but the consensus in our meeting seemed to be leaving this as is for now to prevent confusion and too much churn.